metered/src/metric.rs

```diff
@@ -72,6 +78,15 @@ pub trait Gauge: Default + Clear + Serialize {
     fn decr(&self);
 }
 
+/// A batch trait for Gauges
+pub trait BatchGauge<T>: Gauge {
```

Comment:
I'd like to be able to get rid of the generic T here.  Can we use usize (as a max batch size) and implement proper unroll in the impl side?